doc/GUIDE.md

```diff
@@ -1654,9 +1652,7 @@ the images.
 
 Note that the executable will be built in the development environment
 and copied to the container, so the dev OS must match that of the
-container OS. This is easily accomplished using [Docker integration](docker_integration.md),
-under which the exe emitted by `stack build` will be built on the
-Docker container, not the local OS.
+container OS.
```

Comment:
How about adding that building under Docker with `--docker` or equivalent can also solve this problem?

Reply:
@snoyberg Thanks for the feedback. Sorry that it took so long for me to get back on this.  

The reason I removed this information is because - this is the sentence which caused the confusion for me. While building the code inside the container using `--docker` is valuable, introducing it now itself will confuse the user because the reader as of now is reading about the second integration method ( building code in local machine and copying the executable to the container).  This deleted sentence in the PR confused me enough that I had to trace using `-v` calls in `stack image container` to understand how things work. If you see the paragraph which I added  down in this PR, I explicitly say how to achieve the first integration method and also link a more detailed doc which explains the concept you want me to add here.

If you still want to add, I think adding it below where I explain the difference would be the right place to add. Let me know what you think.

Reply:
On thinking more, I have added about the `--docker` option right below when I start explaining the first integration method. I think that should clarify things.

Reply:
I still think it's insufficiently clear. This is giving the user a warning "you need to match our dev OS to the Docker OS or things will break." It's not immediately obvious that there's a trivial solution to this problem available (building inside Docker too). I think that should be clarified, otherwise we're just dropping a problem in the user's lap without any assistance.

Reply:
@snoyberg Fair enough, I have updated the PR which hopefully makes it more clear.